oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/AnalyzerModuleConfig.java

```diff
@@ -50,7 +51,7 @@
      * Read component-libraries.yml for more details.
      */
     @Getter
-    private String noUpstreamRealAddressAgents = Const.EMPTY_STRING;
+    private final String noUpstreamRealAddressAgents = Const.EMPTY_STRING;
```

Comment:
This value is set by config and system env, so, can't be final.

Reply:
https://github.com/apache/skywalking/blob/1c0fc982b2dc61ca7e452142df7f9846e867d71d/oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/AnalyzerModuleConfig.java#L52-L53

🤔 It seems that we don't modify this filed, and IDE warns us about it. Now I'm not sure. 😂

Reply:
It is not changed directly through codes, it is changed by reflect API of OAP core.

I am checking why you want to change this.